app/renderer/components/tabs/content/favIcon.js

```diff
@@ -47,6 +53,29 @@ class Favicon extends React.Component {
       : this.props.favicon || 'defaultIcon'
   }
 
+  componentDidMount (props) {
+    this.transitionInIfRequired()
+  }
+
+  componentDidUpdate (prevProps) {
+    this.transitionInIfRequired(prevProps)
+  }
+
+  transitionInIfRequired (prevProps) {
+    // transition only if favicon changes
+    const shouldTransitionIn = this.element && !this.props.tabLoading && (!prevProps || this.props.favicon !== prevProps.favicon)
```

Comment:
nit: would be cool/preferable to have a helper method instead or just break this down like 
```js
const shouldTransitionIn = (
  this.element &&
  !this.props.tabLoading &&
  (!prevProps || this.props.favicon !== prevProps.favicon)
)
```